lib/sensu/client/process.rb

```diff
@@ -455,9 +472,11 @@ def resume
       # Stop the Sensu client process, pausing it, completing check
       # executions in progress, closing the transport connection, and
       # exiting the process (exit 0). After pausing the process, the
-      # process/daemon `@state` is set to `:stopping`.
+      # process/daemon `@state` is set to `:stopping`.  Also sends
+      # deregister event if configured to do so.
       def stop
         @logger.warn("stopping")
+        send_deregister_event if @settings[:client][:deregister] == true
         pause
```

Comment:
Wasn't sure about the timing here.  Might be best to put the deregister event after pause so we have a 'guarantee' that no more results/events will be sent.